provider/resource_rediscloud_acl_user.go

```diff
@@ -0,0 +1,145 @@
+package provider
+
+import (
+	"context"
+	"github.com/RedisLabs/rediscloud-go-api/redis"
+	"github.com/RedisLabs/rediscloud-go-api/service/access_control_lists/users"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"strconv"
+	"time"
+)
+
+func resourceRedisCloudAclUser() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Create an ACL User within your Redis Enterprise Cloud Account",
+		CreateContext: resourceRedisCloudAclUserCreate,
+		ReadContext:   resourceRedisCloudAclUserRead,
+		UpdateContext: resourceRedisCloudAclUserUpdate,
+		DeleteContext: resourceRedisCloudAclUserDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(3 * time.Minute),
+			Read:   schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(3 * time.Minute),
+			Delete: schema.DefaultTimeout(1 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "A meaningful name to identify the user",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"role": {
+				Description: "The role which the user has",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"password": {
```

Comment:
I think this'll need a ForceNew, as the value can't be read back from the API so won't be able to spot a drift